### PR TITLE
Add maven task results

### PR DESCRIPTION
**Pull request by valdar**
# Changes
see #1309 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Follows the [authoring recommendations][authoring]
- [x] Includes [docs][docs] (if user facing)
- [X] Includes [tests][tests] (for new tasks or changed functionality)
  - See the [end-to-end testing documentation][e2e] for guidance and CI details.
- [X] Meets the [Tekton contributor standards][contributor] (including functionality, content, code)
- [X] Commit messages follow [commit message best practices][commit]
- [x] Has a kind label. You can add one by adding a comment on this PR that
  contains `/kind <type>`. Valid types are bug, cleanup, design, documentation,
  feature, flake, misc, question, tep
- [X] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [X] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [X] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [X] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [X] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [X] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md) for more details._

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]: https://github.com/tektoncd/catalog/issues/413
[authoring]: https://github.com/tektoncd/catalog/blob/main/recommendations.md
[docs]: https://github.com/tektoncd/community/blob/master/standards.md#docs
[tests]: https://github.com/tektoncd/community/blob/master/standards.md#tests
[e2e]: https://github.com/tektoncd/catalog/blob/main/CONTRIBUTING.md#end-to-end-testing
[contributor]: https://github.com/tektoncd/community/blob/main/standards.md
[commit]: https://github.com/tektoncd/community/blob/master/standards.md#commit-messages

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]

File: task/maven/0.4/maven.yaml [hunk 1]
```diff
@@ -0,0 +1,176 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: maven
+  labels:
+    app.kubernetes.io/version: "0.3"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.17.0"
```
Comment (vinamra28) [inline task/maven/0.4/maven.yaml:8]:
```suggestion
    tekton.dev/pipelines.minVersion: "0.50.0"
```

File: task/maven/0.4/README.md [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+            - package
+      workspaces:
+        - name: maven-settings
+          workspace: maven-settings
+        - name: source
+          workspace: shared-workspace
+        - name: maven-local-repo
+          workspace: maven-local-m2
+---
+apiVersion: tekton.dev/v1beta1
```
Comment (vinamra28) [inline task/maven/0.4/README.md:120]:
```suggestion
apiVersion: tekton.dev/v1
```

File: task/maven/0.4/README.md [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+      persistentvolumeclaim:
+        claimName: maven-local-m2-pvc
+```
+
+---
+
+### With Custom Maven Params
+
+```yaml
+apiVersion: tekton.dev/v1beta1
```
Comment (vinamra28) [inline task/maven/0.4/README.md:143]:
```suggestion
apiVersion: tekton.dev/v1
```

File: task/maven/0.4/maven.yaml [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+        - $(workspaces.maven-settings.path)/settings.xml
+        - "$(params.GOALS)"
+        - '-Dmaven.repo.local=$(workspaces.maven-local-repo.path)/.m2'
+
+    - name: mvn-results
+      image: $(params.MAVEN_IMAGE)
+      workingDir: $(workspaces.source.path)/$(params.CONTEXT_DIR)
+      script: |
+        #!/usr/bin/env bash
+        VERSION=$(/usr/bin/mvn -s $(workspaces.maven-settings.path)/settings.xml -q -Dexec.executable=echo -Dexec.args='${project.version}' --non-recursive exec:exec)
```
[the remark this answers is not in the document]
Comment (valdar) [reply task/maven/0.4/maven.yaml:171]: We sure can, I decided do have them separate as a logical division between the two but I was concerned about execution time as well.

[the remark this answers is not in the document]
Comment (valdar) [reply task/maven/0.4/maven.yaml]: see if the proposed changes makes sense to you

**Review by vinamra28 — APPROVED** [review]
/lgtm

**Pull request merged**